packages/backend-api/src/Controller/V1/Product/ProductController.php

```diff
@@ -0,0 +1,285 @@
+        $this->disallowWorkingWithVariants($product);
+
+        $this->productFacade->delete($product->getId());
+
+        $view = View::create([], Response::HTTP_NO_CONTENT);
+
+        return $this->handleView($view);
+    }
+
+    /**
+     * @param array $errors
+     * @return \FOS\RestBundle\View\View
+     */
+    protected function createValidationView(array $errors): View
+    {
+        $code = Response::HTTP_BAD_REQUEST;
+        $message = 'Provided data did not pass validation';
+
+        return View::create(['code' => $code, 'message' => $message, 'errors' => $errors], $code);
+    }
+
+    /**
+     * @param array $uuids
+     */
+    protected function validateUuids(array $uuids): void
+    {
+        $invalidUuids = [];
+
+        foreach ($uuids as $uuid) {
+            if (!Uuid::isValid($uuid)) {
+                $invalidUuids[] = $uuid;
+            }
+        }
+
+        if (count($invalidUuids) === 1) {
+            throw new BadRequestHttpException('This UUID is not valid: ' . reset($invalidUuids));
+        } elseif (count($invalidUuids) > 1) {
+            throw new BadRequestHttpException('These UUIDS are not valid: ' . implode(', ', $invalidUuids));
+        }
+    }
+
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param string|null $uuid
+     * @return array
+     */
+    protected function validateCreateProductInputs(Request $request, ?string $uuid): array
+    {
+        $errors = $this->productApiDataValidator->validateCreate($request->request->all());
+
+        if ($uuid !== null) {
+            if (!Uuid::isValid($uuid)) {
+                $errors['[uuid]'] = 'This UUID is not valid: ' . $uuid;
+            } else {
+                try {
+                    $this->productFacade->getByUuid($uuid);
+
+                    throw new UnprocessableEntityHttpException('Product with ' . $uuid . ' UUID already exists');
+                } catch (\Shopsys\FrameworkBundle\Model\Product\Exception\ProductNotFoundException $e) {
+                }
+            }
+        }
+
+        return $errors;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Product\Product $product
+     */
+    protected function disallowWorkingWithVariants(Product $product): void
+    {
+        if ($product->isVariant() || $product->isMainVariant()) {
+            throw new BadRequestHttpException('cannot update variant/main variant, this functionality is not supported yet');
+        }
+    }
+
+    /**
+     * @Patch("/products/{uuid}")
+     * @param string $uuid
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function patchProductAction(string $uuid, Request $request): Response
+    {
+        $this->validateUuids([$uuid]);
+        $productApiData = $request->request->all();
+        $errors = $this->productApiDataValidator->validateUpdate($productApiData);
+        if (count($errors)) {
```

Comment:
please be explicit with conditions
```suggestion
        if (count($errors) > 0) {
```

Reply:
Really? I personally find the sorter version better, sort of idiom

Reply:
Well, I like being explicit :) 
But as we don't have any agreement about this topic in our team it's ok.